NMS-8569: Provisiond doc (#5172)
OpenNMS/opennms: docs/modules/reference/pages/daemons/daemon-config-files/provisiond.adoc

[[ref-daemon-config-files-provisiond]]
= Provisiond

The provision daemon handles scanning of requisition data to xref:operation:provisioning/introduction.adoc[provision] into monitored nodes.

It is recommended that you update configuration settings using the {page-component-title} UI.

.Provisiond configuration file overview
[options="header"]
[cols="2,3,1,1"]
|===
| File
| Description
| Reload Event
| Restart Required

ifeval::["{page-component-title}" == "Meridian"]
| provisiond-configuration.xml
| This file sets requisition definitions for defined discovery of nodes and foreign sources.
| yes
| yes
endif::[]

> | foreign-sources.xsd
> | Settings for provisioning foreign sources. +
> If these settings are modified using the UI, a `default-foreign-sources.xsd` file will be automatically created in the root directory.

Reviewer: ```suggestion
| default-foreign-source.xml
| List of xref:reference:provisioning/detectors.adoc[detectors] and xref:reference:provisioning/policies.adoc[policies] to apply when scanning requisitioned nodes.
Does not exist by default.
Once these settings are modified using the UI, a `default-foreign-source.xml` file will be automatically created in `$\{OPENNMS_HOME}/etc`.
```

| no
| no

> |===

Reviewer: ```suggestion

| foreign-sources/*.xml
| List of xref:reference:provisioning/detectors.adoc[detectors] and xref:reference:provisioning/policies.adoc[policies] to apply when scanning nodes in a requisition matching the filename.
Does not exist by default.
| yes
| no

| imports/*.xml
| One file per requisition, containing the list of nodes to be imported for monitoring.
| yes
| no
|===
```

[[ref-daemon-config-files-provisiond]]
= Provisiond

> The provision daemon handles scanning of requisition data to xref:operation:provisioning/introduction.adoc[provision] into monitored nodes.

Reviewer 2: This could be a case of "provision" meaning too many things, but I don't think this is quite the process. It sounds like the provisioning is happening into the monitored nodes, whereas I think of provisioning as identifying the nodes you want to monitor in Horizon ("importing" the nodes).

Not sure what to suggest on Friday afternoon. :-)

Author: @mmahacek Would you have a suggestion about how to clarify this sentence? :)

Reviewer: Provisiond scans nodes a) from requisition/import files and b) from newSuspect events to import into the database. Only nodes in the database are monitored by other daemons. 

NewSuspect events are triggered by discover scans and by trapd/syslogd if those daemons are configured to create the event if an incoming message doesn't match an existing node.

Author: Thanks!

```suggestion
The provision daemon scans nodes from requisition and import files, and from newSuspect events.
The scanned nodes are then added to the database.
Note that only nodes in the database are monitored by other daemons.

NewSuspect events are triggered by Discover scans, and by trapd and syslogd if those daemons are configured to create the event.
When configured properly, trapd and syslogd create a newSuspect event if an incoming message does not match any existing nodes.
```

[[ref-daemon-config-files-provisiond]]
= Provisiond

The provision daemon scans nodes from requisition and import files, and from newSuspect events.
The scanned nodes are then added to the database.
Note that only nodes in the database are monitored by other daemons.

> NewSuspect events are triggered by Discover scans, and by trapd and syslogd if those daemons are configured to create the event.

Reviewer: ```suggestion
NewSuspect events are triggered by Discovery scans, and by trapd and syslogd if those daemons are configured to create the event.
```

> When configured properly, trapd and syslogd create a newSuspect event if an incoming message does not match any existing nodes.

Reviewer: This line is redundant to the previous sentence.

Author: ```suggestion
```
